SRS/SRS_GRIDDED/backlog_pipeline_reprocessing.py

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+"""
+Script to copy and reprocess n_files_limit(default is 20) GHRSST files
+via pipeline in order not to overflow the celery queue with backlog items to
+reprocess. New files being uploaded by the facility will still be processed
+within a reasonable time frame of couple of minutes max
+
+This script should be run in a tmux/screen/background as it will take a long
+long time to run
+roughly 770000 files (10 sec/file) = 3 months of reprocessing
+
+default input file is list_ghrsst_files.tgz . But any text file can be parsed
+* $WIP_DIR/SRS/backlog_files_not_compliant contains a list of files not
+compliant, so not pushed back yet. They need to be modified and re-uploaded by
+the facility
+* $WIP_DIR/SRS/backlog_ghrsst_files_already_reprocessed_to_pipeline
+
+
+the list_ghrsst_files.tgz was generated AFTER new files already made it through
+the generic timestep harvester. This meens that once this script finishes to
+run, the only backlog left will be files not passing the ghrsst/cf checker
+"""
+
+import argparse
+import errno
+import os
+import shutil
+import tarfile
+import tempfile
+
+from util import pass_netcdf_checker
+
+
+def mkdir_p(path):
+    try:
+        os.makedirs(path)
+    except OSError as exc:  # Python >2.5
+        if exc.errno == errno.EEXIST and os.path.isdir(path):
+            pass
+        else:
+            raise
+
+
+def list_files(fname):
+    "read content of text file into list. handle tgz file (smaller for github)"
+    if fname.endswith('.tgz'):
+        tar = tarfile.open(fname, mode="r")
```

Comment:
If this only ever holds a single file, it should probably be a straight GZ rather than tar.gz. Then you could just stream the content out without the mucking around with temp dirs and cleanup?